kernel/api/src/main/java/org/sakaiproject/user/api/CandidateDetailProvider.java

```diff
@@ -0,0 +1,43 @@
+package org.sakaiproject.user.api;
+
+import org.sakaiproject.site.api.Site;
+import org.sakaiproject.user.api.User;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * This is a provider interface that allows Assignments to provide addition details about candidates
+ * to the interface.
+ */
+public interface CandidateDetailProvider {
+	/**
+	 * This gets an candidate ID for a user, this can be used to make candidate IDs anonymous.
+	 * @param user The user for who an ID is wanted. Cannot be <code>null</code>
+	 * @param site The site in which the lookup is happening. If site is null, it will try to get the current site
```

Comment:
Having APIs that rely on data set in the thread by the portal isn't good as it makes supporting them through /direct more complicated and error prone. Can we make the site object required?